sdk/metric/meter_example_test.go

```diff
@@ -0,0 +1,371 @@
+		Scope: instrumentation.Scope{Name: "AsynchronousExample"},
+		Metrics: []metricdata.Metrics{
+			{
+				Name: "filteredPageFaults",
+				Data: metricdata.Sum[int64]{
+					Temporality: metricdata.DeltaTemporality,
+					IsMonotonic: true,
+					DataPoints: []metricdata.DataPoint[int64]{
+						{
+							Attributes: filteredAttributeSet,
+							Value:      80,
+						},
+					},
+				},
+			},
+			{
+				Name: "pageFaults",
+				Data: metricdata.Sum[int64]{
+					Temporality: metricdata.DeltaTemporality,
+					IsMonotonic: true,
+					DataPoints: []metricdata.DataPoint[int64]{
+						{
+							Attributes: attribute.NewSet(tid1Attrs...),
+							Value:      50,
+						},
+						{
+							Attributes: attribute.NewSet(tid2Attrs...),
+							Value:      30,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	metrics, err := reader.Collect(ctx)
+	assert.NoError(t, err)
+	metricdatatest.AssertEqual(t, wantScopeMetrics, metrics.ScopeMetrics[0], metricdatatest.IgnoreTimestamp())
+
+	wantFilterValue := &wantScopeMetrics.Metrics[0].Data.(metricdata.Sum[int64]).DataPoints[0].Value
+	wantDataPoint1Value := &wantScopeMetrics.Metrics[1].Data.(metricdata.Sum[int64]).DataPoints[0].Value
+	wantDataPoint2Value := &wantScopeMetrics.Metrics[1].Data.(metricdata.Sum[int64]).DataPoints[1].Value
+
+	// During the time range (T1, T2]:
+	//     pid = 1001, tid = 1, #PF = 53
+	//     pid = 1001, tid = 2, #PF = 38
+
+	atomic.StoreInt64(&pfValues[0], 53)
+	atomic.StoreInt64(&pfValues[1], 38)
+
+	*wantFilterValue = 11
+	*wantDataPoint1Value = 3
+	*wantDataPoint2Value = 8
+
+	metrics, err = reader.Collect(ctx)
+	assert.NoError(t, err)
+	metricdatatest.AssertEqual(t, wantScopeMetrics, metrics.ScopeMetrics[0], metricdatatest.IgnoreTimestamp())
+
+	// During the time range (T2, T3]
+	//     pid = 1001, tid = 1, #PF = 56
+	//     pid = 1001, tid = 2, #PF = 42
+
+	atomic.StoreInt64(&pfValues[0], 56)
+	atomic.StoreInt64(&pfValues[1], 42)
+
+	*wantFilterValue = 7
+	*wantDataPoint1Value = 3
+	*wantDataPoint2Value = 4
+
+	metrics, err = reader.Collect(ctx)
+	assert.NoError(t, err)
+	metricdatatest.AssertEqual(t, wantScopeMetrics, metrics.ScopeMetrics[0], metricdatatest.IgnoreTimestamp())
+
+	// During the time range (T3, T4]:
+	//     pid = 1001, tid = 1, #PF = 60
+	//     pid = 1001, tid = 2, #PF = 47
+
+	atomic.StoreInt64(&pfValues[0], 60)
+	atomic.StoreInt64(&pfValues[1], 47)
+
+	*wantFilterValue = 9
+	*wantDataPoint1Value = 4
+	*wantDataPoint2Value = 5
+
+	metrics, err = reader.Collect(ctx)
+	assert.NoError(t, err)
+	metricdatatest.AssertEqual(t, wantScopeMetrics, metrics.ScopeMetrics[0], metricdatatest.IgnoreTimestamp())
+
+	// During the time range (T4, T5]:
+	//     thread 1 died, thread 3 started
+	//     pid = 1001, tid = 2, #PF = 53
+	//     pid = 1001, tid = 3, #PF = 5
+
+	atomic.StoreInt64(&pfValues[0], 0)
+	atomic.StoreInt64(&pfValues[1], 53)
+	atomic.StoreInt64(&pfValues[2], 5)
+
+	*wantFilterValue = -49 // TODO: This should be 11
```

Comment:
Why should this be `11` not `-49`? In the cumulative example it went from `107` to `58` which is a delta of `-49`.

Reply:
```suggestion
	*wantFilterValue = -49
```